src/saltext/vmware/states/vmc_dhcp_profiles.py

```diff
@@ -0,0 +1,390 @@
+
+    org_id
+        The Id of organization to which the SDDC belongs to
+
+    sddc_id
+        The Id of SDDC for which the DHCP profile should be added
+
+    type
+        The type of DHCP profile for which the given dhcp belongs to. Possible values: server, relay
+
+    dhcp_profile_id
+        DHCP profile id, any static unique string identifying the DHCP profile.
+        Also same as the display_name by default.
+
+    verify_ssl
+        (Optional) Option to enable/disable SSL verification. Enabled by default.
+        If set to False, the certificate validation is skipped.
+
+    cert
+        (Optional) Path to the SSL client certificate file to connect to VMC Cloud Console.
+        The certificate can be retrieved from browser.
+
+    server_addresses
+        when type is relay, this field indicates DHCP relay addresses(DHCP server IP addresses
+        for DHCP relay configuration).
+        Both IPv4 and IPv6 addresses are supported.
+
+        when type is server, this field indicates DHCP server address in CIDR format.
+        Both IPv4 and IPv6 address families are supported.
+        Prefix length should be less than or equal to 30 for IPv4 address family
+        and less than or equal to 126 for IPv6.
+        When not specified, IPv4 value is auto-assigned to 100.96.0.1/30.
+
+        Note: This field is optional only when type is server and
+        is mandatory when type is relay.
+
+    tags
+        (Optional) Opaque identifiers meaningful to the user.
+
+        .. code::
+
+            tags:
+              - tag: <tag-key-1>
+                scope: <tag-value-1>
+              - tag: <tag-key-2>
+                scope: <tag-value-2>
+
+    lease_time
+        (Optional) IP address lease time in seconds. Minimum value is 60.
+        Maximum value is 4294967295. Default value is 86400
+        Note: This field is applicable only when type is server
+
+    display_name
+        Identifier to use when displaying entity in logs or GUI.
+
+    Example values:
+
+        .. code::
+
+            for dhcp-server-profiles:
+
+            server_addresses:
+              - 10.22.12.2/23
+            tags:
+              - tag: tag1
+                scope: scope1
+            lease_time: 86400
+
+            for dhcp-relay-profiles:
+
+            server_addresses:
+              - 10.1.1.1
+            tags:
+              - tag: tag1
+                scope: scope1
+
+    """
+
+    input_dict = {
+        "server_addresses": server_addresses,
+        "tags": tags,
+        "lease_time": lease_time,
+        "display_name": display_name,
+    }
+
+    if type == vmc_constants.RELAY and lease_time:
+        return vmc_state._create_state_response(
+            name=name, comment="lease_time is not applicable for DHCP Relay Profile", result=False
+        )
+
+    if type == vmc_constants.RELAY:
+        input_dict.pop("lease_time")
+
+    input_dict = {k: v for k, v in input_dict.items() if v != vmc_constants.VMC_NONE}
```

Comment:
**suggestion** altogether this would be much better written:


```
# This could go in utils, since it's probably more broadly applicable

def _filter_vmc_none(**kwargs):
    return {k: v for k,v in kwargs.items() if v != vmc_constants.VMC_NONE}
```
```
input_dict = {}
if type == vmc_constants.RELAY:
    if lease_time:
        return ...
else:
    if lease_time != vmc_constants.VMC_NONE:
        input_dict['lease_time'] = lease_time

input_dict.update(_filter_vmc_none(
        server_addresses=server_addresses,
        tags=tags,
        display_name=display_name,
        ))
```

Reply:
updated